docs/src/inverse_problems/optimization_ode_param_fitting.md

```diff
@@ -0,0 +1,173 @@
+plot!(fitted_sol_S_P; idxs=[:S, :P], label="Fitted solution", linestyle=:dash, lw=6, color=[:lightblue :pink])
+```
+
+## Setting parameter constraints and boundaries
+Sometimes, it is desired to set boundaries on parameter values. Indeed, this can speed up the optimisation process (by preventing searching through unfeasible parts of parameter space) or be a prerequisite of some optimisation methods. This can be done by passing the `lb` (lower bounds) and `up` (upper bounds) arguments to `OptimizationProblem`. These are vectors (of the same length as the number of parameters), with each argument corresponding to the boundary value of the parameter with the same index (as used in the `parameters(rn)` vector). If we wish to constrain each parameter to the interval *(0.1, 10.0)* this can be done through:
+```@example diffeq_param_estim_1 
+optprob = OptimizationProblem(loss_function, [1.0, 1.0, 1.0]; lb = [0.1, 0.1, 0.1], ub = [10.0, 10.0, 10.0])
+nothing # hide
+```
+
+In addition to boundaries, Optimization.jl also supports setting [linear and non-linear constraints](https://docs.sciml.ai/Optimization/stable/tutorials/constraints/#constraints) on its output solution.
+
+## Parameter fitting with known parameters
+If we from previous knowledge know that *kD = 0.1*, and only would like to fit the values of *kD* and *kP*, this can be achieved through `build_loss_objective`'s `prob_generator` argument. First, we create a function (`fixed_p_prob_generator`) that modifies our `ODEProblem` to incorporate this knowledge:
+```@example diffeq_param_estim_1 
+fixed_p_prob_generator(prob, p) = remake(prob; p = vcat(p[1], 0.1, p[2]))
+nothing # hide
+```
+Here, it takes the `ODEProblem` (`prob`) we simulates, and the parameter set used, during the optimisation process (`p`), and creates a modified `ODEProblem` (by setting a customised parameter vector [using `remake`](@ref simulation_structure_interfacing_remake)). Now we create our modified loss function:
+```@example diffeq_param_estim_1 
+loss_function_fixed_kD = build_loss_objective(oprob, Tsit5(), L2Loss(data_ts, data_vals), Optimization.AutoForwardDiff(); prob_generator = fixed_p_prob_generator, maxiters=10000, verbose=false, save_idxs=4)
+nothing # hide
+```
+
+We can create an optimisation problem from this one like previously, but keeping in mind that it (and its output results) only contains two parameter values (*kB* and *kP):
+```@example diffeq_param_estim_1 
+optprob_fixed_kD = OptimizationProblem(loss_function_fixed_kD, [1.0, 1.0])
+optsol_fixed_kD = solve(optprob_fixed_kD, Optim.NelderMead())
+nothing # hide
+```
+
+## Fitting parameters on the logarithmic scale
+Often it can be advantageous to fit parameters on a [logarithmic, rather than linear, scale](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008646). The best way to proceed is to simply replace each parameters in the model definition by its logarithmic version:
```

Comment:
```suggestion
Often it can be advantageous to fit parameters on a [logarithmic, rather than linear, scale](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008646). The best way to proceed is to simply replace each parameter in the model definition by its logarithmic version:
```
This doesn't seem like the right approach. One should just build a loss function wrapper that does the conversion for you instead of changing the model definition of parameters. This should be a trivial transformation to implement.

Reply:
OK. It probably is trivial, but I have to admit I don't understand how to do it from your example. It sounds like the best solution is to make an example problem where does that. put that under example problems, and then reference that.

Generally, PEtab handling this kind of stuff automatically (if you want) is one of the advantages of just using that one.